codegen/src/api/calls.rs

```diff
@@ -21,74 +21,63 @@ use subxt_metadata::PalletMetadata;
 pub fn generate_calls(
     type_gen: &TypeGenerator,
     pallet: &PalletMetadata,
-    types_mod_ident: &syn::Ident,
     crate_path: &syn::Path,
-    should_gen_docs: bool,
 ) -> Result<TokenStream2, CodegenError> {
     // Early return if the pallet has no calls.
     let Some(call_ty) = pallet.call_ty_id() else {
         return Ok(quote!());
     };
 
-    let mut struct_defs = super::generate_structs_from_variants(
+    let variant_names_and_struct_defs = super::generate_structs_from_variants(
         type_gen,
-        types_mod_ident,
         call_ty,
         |name| name.to_upper_camel_case().into(),
         "Call",
-        crate_path,
-        should_gen_docs,
     )?;
-
-    let result = struct_defs
-        .iter_mut()
-        .map(|(variant_name, struct_def, aliases)| {
-            let fn_name = format_ident!("{}", variant_name.to_snake_case());
-
-            let result: Vec<_> = match struct_def.fields {
-                CompositeDefFields::Named(ref named_fields) => named_fields
+    let (call_structs, call_fns): (Vec<_>, Vec<_>) = variant_names_and_struct_defs
+        .into_iter()
+        .map(|var| {
+            let (call_fn_args, call_args): (Vec<_>, Vec<_>) = match &var.composite.kind {
+                CompositeIRKind::Named(named_fields) => named_fields
                     .iter()
                     .map(|(name, field)| {
-                        let call_arg = if field.is_boxed() {
+                        // Note: fn_arg_type this is relative the type path of the type alias when prefixed with `types::`, e.g. `set_max_code_size::New`
```

Comment:
Could there be a variant of the metadata where this does not correspond to the expected alias? Or it would be different from the previous `format_ident!("{}", name.to_string().to_upper_camel_case())`?

Reply:
Good question, when calling `generate_structs_from_variants`, which calls `generate_type_alias_mod` for each such composite, for each field, the field.type_path is replaced by `#alias_mod_name::#alias_name`, where `alias_name` is either `format_ident!("{}", name.to_string().to_upper_camel_case())` (named fields) or `format_ident!("Field{}", i)` for unnamed fields, so the logic should be very similar to what you implemented before. I cannot think of a case right now where this is not fulfilled.